tests/7zip_compatiblity_test.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg(feature = "test-7zip")]
+//! This test compresses some random data with deflate64 using p7zip `7z` command and check decompression
+
+use std::ffi::OsString;
+use std::fs::File;
+use deflate64::Deflate64Decoder;
+use std::io::{Cursor, Read, Write};
+use std::mem::size_of;
+use std::process::{Command, Stdio};
+use proptest::proptest;
+use tempfile::TempDir;
+
+const TEST_FILE_NAME: &'static str = "test.file";
+const TEST_ZIP_NAME: &'static str = "test.zip";
+
+#[repr(packed)]
+struct ZipLocalFileHeader {
+    signature: [u8; 4],
+    version: [u8; 2],
+    flags: [u8; 2],
+    compression_method: [u8; 2],
+    last_mod_time: [u8; 2],
+    last_mod_date: [u8; 2],
+    uncompressed_crc32: [u8; 4],
+    compressed_size: [u8; 4],
+    uncompressed_size: [u8; 4],
+    file_name_len: [u8; 2],
+    extra_field_len: [u8; 2],
+}
+
+impl ZipLocalFileHeader {
+    const SIGNATURE: [u8; 4] = [0x50, 0x4B, 0x03, 0x04];
+
+    fn zero() -> Self {
+        unsafe { std::mem::zeroed() }
+    }
+
+    fn as_mut_bytes(&mut self) -> &mut [u8] {
+        unsafe {
+            std::slice::from_raw_parts_mut(
+                self as *mut Self as *mut u8,
+                size_of::<Self>()
+            )
+        }
+    }
+}
+
+fn compress_with_7zip(data: &[u8]) -> Vec<u8> {
+    let mut temp_dir = TempDir::new().unwrap();
+
+    // write data to test file
+    File::create(temp_dir.path().join(TEST_FILE_NAME)).unwrap()
+        .write_all(data).unwrap();
+
+    let seven_zip = std::env::var_os("SEVEN_ZIP_PATH").unwrap_or_else(|| OsString::from("7z"));
+
+    let mut seven_zip_process = Command::new(seven_zip)
+        .arg("a")
+        .arg("-mm=Deflate64")
+        .arg(TEST_ZIP_NAME)
+        .arg(TEST_FILE_NAME)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdin(Stdio::null())
```

Comment:
[`Command::output`](https://doc.rust-lang.org/std/process/struct.Command.html#method.output) can be used to simplify this.